sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/xml/StaxXmlParserUtils.scala

```diff
@@ -178,4 +179,11 @@ object StaxXmlParserUtils {
       }
     }
   }
+
+  def consumeNextEndElement(parser: XMLEventReader): Unit = {
```

Comment:
For defensive purpose, consider passing the name of the `StartElement` and assert that it matches with the `EndElement`:

```suggestion
  def skipNextEndElement(parser: XMLEventReader): Unit = {
```

Reply:
Added an assertion on the endElementName